src/StargateEventListener/Handlers/StargateEventHandlers.ts

```diff
@@ -0,0 +1,269 @@
+    nodeId: string
+    delegatee: string
+    isDelegated: boolean
+    blockNumber: number
+    txId: string
+    contractAddress: string
+}
+
+interface NodeDelegatedEventHandlerProps {
+    beat: Beat
+    network: Network
+    thor: ThorClient
+    refetchStargateData: () => void
+}
+
+export const handleNodeDelegatedEvent = async ({
+    beat,
+    network,
+    thor,
+    refetchStargateData,
+}: NodeDelegatedEventHandlerProps) => {
+    try {
+        // Get Stargate contract addresses for the current network
+        const networkConfig = getStargateNetworkConfig(network.type)
+
+        if (!networkConfig) {
+            debug(ERROR_EVENTS.STARGATE, `No Stargate configuration found for network: ${network.type}`)
+            return
+        }
+
+        const stargateContracts = [
+            networkConfig.NODE_MANAGEMENT_CONTRACT_ADDRESS,
+            networkConfig.STARGATE_DELEGATION_CONTRACT_ADDRESS,
+            networkConfig.STARGATE_NFT_CONTRACT_ADDRESS,
+            networkConfig.LEGACY_NODES_CONTRACT_ADDRESS,
+        ].filter(Boolean)
+
+        // Check if beat might contain Stargate events using bloom filter
+        const hasStargateContract = stargateContracts.some(contractAddress =>
+            BloomUtils.testBloomForAddress(beat.bloom, beat.k, contractAddress),
+        )
+
+        if (!hasStargateContract) {
+            return // No Stargate contracts in this beat
+        }
+
+        debug(ERROR_EVENTS.STARGATE, `Potential Stargate events in beat ${beat.number}`)
+
+        // Parse NodeDelegated events from the beat (specifically from NodeManagement contract)
+        const nodeManagementAddress = networkConfig.NODE_MANAGEMENT_CONTRACT_ADDRESS
+        if (!nodeManagementAddress) {
+            debug(ERROR_EVENTS.STARGATE, `No NodeManagement contract address found for network: ${network.type}`)
+            return
+        }
+
+        const nodeDelegatedEvents = await parseNodeDelegatedEvents(beat, nodeManagementAddress, thor)
+
+        if (nodeDelegatedEvents.length > 0) {
+            debug(
+                ERROR_EVENTS.STARGATE,
+                `Found ${nodeDelegatedEvents.length} NodeDelegated events in beat ${beat.number}`,
+            )
+
+            // Process each event
+            for (const eventData of nodeDelegatedEvents) {
+                await processNodeDelegatedEvent(eventData, refetchStargateData)
+            }
+        }
+    } catch (err) {
+        error(ERROR_EVENTS.STARGATE, "Error handling NodeDelegated event:", err)
+    }
+}
+
+/**
+ * Parse NodeDelegated events from a beat
+ */
+const parseNodeDelegatedEvents = async (
+    beat: Beat,
+    nodeManagementContract: string,
+    thor: ThorClient,
+): Promise<NodeDelegatedEventData[]> => {
+    try {
+        const block = await thor.blocks.getBlockCompressed(beat.id)
+
+        if (!block?.transactions) {
+            debug(ERROR_EVENTS.STARGATE, `No transactions found in block ${beat.number}`)
+            return []
+        }
```

Comment:
I'd suggest getting the expanded block, to avoid getting the receipt for every tx